modules/core/src/main/scala/com/azavea/stac4s/StacItemAsset.scala

```diff
@@ -1,27 +1,63 @@
 package com.azavea.stac4s
 
 import cats.Eq
+import cats.implicits._
 import io.circe._
+import io.circe.syntax._
 
 final case class StacItemAsset(
     href: String,
     title: Option[String],
     description: Option[String],
     roles: List[StacAssetRole],
```

Comment:
Not relevant for this PR necessarily, but since this is being worked on I think this should be a `Set[StacAssetRole]`.

The only reason I bring this up is because when I followed your testing instructions and looked at the resulting collection with items I noticed this for an asset...

```
"roles" : [                               
        "host",                                               
        "processor",                            
        "processor",                            
        "processor",                            
        "processor",                            
        "processor",                            
        "producer",                             
        "host",                                 
        "processor",                            
        "licensor",                             
        "licensor",                             
        "processor",                                   
        "producer",                             
        "producer",                                    
        "licensor",                             
        "processor",                            
        "producer",                             
        "processor",                            
        "processor",                                          
        "host",                                               
        "processor",
        "licensor",
        "licensor",
        "licensor",
        "host",
        "licensor",
        "licensor",
        "licensor",
        "licensor",
        "host",
        "host",
        "licensor",
        "licensor",
        "host",
        "producer",
        "processor",
        "licensor",
        "licensor",
        "processor",
...
```

It seems like we would want to reject data from re-using the same role multiple times.